packer/plugin.go

```diff
@@ -268,4 +269,26 @@ type PluginDetails struct {
 	PluginPath  string
 }
 
-var PluginsDetailsStorage = map[string]PluginDetails{}
+type PluginsDetailsStorage struct {
+	rwMutex sync.RWMutex
+	data    map[string]PluginDetails
+}
+
+var GlobalPluginsDetailsStore = &PluginsDetailsStorage{
+	data: make(map[string]PluginDetails),
```

Comment:
I advise we instanciate the mutex here, does locking/unlocking work with a nil receiver though?
```suggestion
	rwMutex sync.RWMutex
	data:   make(map[string]PluginDetails),
```

Reply:
So as it happens to be the zero value for a RWMutex is an unlocked mutex. No need to initialize 🤔

Reply:
You can skip the mutex initialization. By default zero value is the unlocked mutex.